platform-api/src/main/java/io/datacater/core/deployment/DeploymentEndpoint.java

```diff
@@ -133,6 +135,33 @@ public Uni<Response> getMetrics(
                 }));
   }
 
+  @GET
+  @Path("{uuid}/status")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Uni<DataCaterDeploymentStatus> getStatusByUuid(@PathParam("uuid") UUID deploymentId) {
+    return dsf.withTransaction(
+            ((session, transaction) -> session.find(DeploymentEntity.class, deploymentId)))
+        .onItem()
+        .ifNull()
+        .failWith(new DeploymentNotFoundException(StaticConfig.LoggerMessages.DEPLOYMENT_NOT_FOUND))
+        .onItem()
+        .ifNotNull()
+        .transform(
+            deploymentEntity -> {
+              try {
+                Deployment deployment = new K8Deployment(client).getDeploymentObject(deploymentId);
+                return DataCaterDeploymentStatus.from(deployment);
+              } catch (KubernetesClientException e) {
+                LOGGER.warn(StaticConfig.LoggerMessages.DEPLOYMENT_NOT_FOUND, e);
+                return null;
+              }
+            })
+        .onItem()
+        .ifNull()
+        .failWith(
+            new DeploymentNotFoundException(StaticConfig.LoggerMessages.DEPLOYMENT_NOT_FOUND));
+  }
```

Comment:
Thank you for your feedback! You are right, this does not look very clean. I changed this part, so that if a `KubernetesClientException` happens, an unchecked `DeploymentNotFoundException` is thrown. This way I could get rid of the entire `return null` and `.onItem().ifNull().failWith()` part. Please lmk what you think